File: apps/backend/app/models/address.py
# backend/app/models/address.py

from uuid import uuid4
from sqlalchemy import Column, String, ForeignKey, Boolean
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from sqlalchemy import DateTime
from sqlalchemy.orm import relationship
from app.db.base import Base

class Address(Base):
    __tablename__ = "addresses"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid4)
    user_id = Column(
        UUID(as_uuid=True),
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )

    # Campos obrigatórios básicos
    street = Column(String(255), nullable=False)
    city = Column(String(100), nullable=False)
    state = Column(String(100), nullable=False)
    postal_code = Column(String(20), nullable=False)
    country = Column(String(100), nullable=False, default="Brasil")

    # Novos campos:
    number = Column(String(20), nullable=False)               # número da residência
    neighborhood = Column(String(100), nullable=False)         # bairro
    complement = Column(String(255), nullable=True)            # complemento opcional
    is_selected = Column(Boolean, default=False, nullable=False)  # se este é o endereço “principal”

    created_at = Column(DateTime(timezone=True), server_default=func.now())

    # Relação inversa opcional (endereço → usuário)
    user = relationship("User", back_populates="addresses")

File: apps/portal-client/src/types/address.ts
// src/types/address.ts

export interface AddressBase {
  street: string;
  number: string;
  neighborhood: string;
  complement?: string;
  city: string;
  state: string;
  postal_code: string;
  country?: string; // padrão "Brasil"
  is_selected?: boolean; // opcional no front, default=false no back
}

// Na criação, `is_selected` existe mas pode ficar false por padrão
export interface AddressCreate extends AddressBase {
  is_selected?: boolean;
}
export interface AddressRead extends AddressBase {
  id: string;
  user_id: string;
  is_selected: boolean;
  created_at: string;
}

export interface AddressUpdate {
  street?: string;
  number?: string;
  neighborhood?: string;
  complement?: string;
  city?: string;
  state?: string;
  postal_code?: string;
  country?: string;
  is_selected?: boolean;
}